src/__internal__/fieldset/fieldset.d.ts

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+
+export interface FieldsetProps {
+  /** Fieldset content */
+  children: React.ReactNode;
+  /** The content for the Fieldset Legend */
+  legend?: string;
+  /* Indicate that error has occurred
+  Pass string to display icon, tooltip and red border
+  Pass true boolean to only display red border */
+  error?: boolean | string;
+  /* Indicate that warning has occurred
+  Pass string to display icon, tooltip and orange border
+  Pass true boolean to only display orange border */
+  warning?: boolean | string;
+  /* Indicate additional information
+  Pass string to display icon, tooltip and blue border
+  Pass true boolean to only display blue border */
+  info?: boolean | string;
+  /** When true, legend is placed in line with the children */
+  inline?: boolean;
+  /** Allows to override existing component styles */
+  styleOverride?: {
+    root?: object;
+    legend?: object;
+  };
+}
+
+declare const Fieldset: React.ComponentClass<FieldsetProps>;
+
+export { Fieldset };
```

Comment:
Could you also add an `index.d.ts` for this component please?

Reply:
Good spot btw